solr/contrib/ltr/src/test/org/apache/solr/ltr/feature/impl/TestExternalFeatures.java

```diff
@@ -115,9 +115,8 @@ public void testEfiFeatureExtraction() throws Exception {
 
     // Using nondefault store should still result in error with no efi
     query.remove("fl");
-    query.add("fl", "[fv store=fstore2]");
-    assertJQ("/query" + query.toQueryString(), "/error/msg=='Exception from createWeight for Feature [name=confidence, type=ValueFeature, id=0, params={value=${myconf}}] org.apache.solr.ltr.feature.impl.ValueFeature.ValueFeatureWeight requires efi parameter that was not passed in request.'");
-
+    query.add("fl", "fvalias:[fv store=fstore2]");
+    assertJQ("/query" + query.toQueryString(), "/response/docs/[0]/fvalias=='originalScore:0.0'");
```

Comment:
Should have both tests, once for the required efi feature, and one for the optional efi feature